### PR TITLE
Add User FAQ

### PR DESCRIPTION
**Pull request by michaelgugino**

### CODE REVIEW

File: FAQ.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+“VM” or “Instance” means the actual compute resource running on the cloud.
+
+“Node” means the kubelet and the “node” object in the API.
+
+These terms, while obviously related, are not interchangeable.  Oftentimes, people will say “I deleted the node” and what they really mean is “I deleted the VM using the cloud provider’s CLI tool.”  As you can see, this can cause confusion and delay in resolving your issue.
+
+# Machines
+
+## Machine Phases?
+Please see docs for machine phases: <link goes here>
```
Comment (michaelgugino) [inline FAQ.md]: Do we have docs for this?

File: FAQ.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+This is not recommended.  By default, the machine-api will not take any corrective action.  If you are utilizing MachineHealthChecking, the machine may get deleted depending on the configuration of the MHC.
+
+## Can I remove the finalizer for a machine that is stuck in deleting?
+This is not recommended.  This may result in orphaned node objects and orphaned compute resources.
+
+## How do I set a Node’s Role (eg, Worker)
+This is set via the MCO’s config, not the machine-api. The config a machine receives upon boot is decided by the machine's user-data.
+
+## How does the user-data get created for the VMs?
+The user-data is generated by the installer.  The machine-api is just a consumer of the user-data and simply passes the data to the cloud.  Refer to documentation for the machine-config-operator (not part of the machine-api) for creating new machine pools and user-data.
```
Comment (michaelgugino) [inline FAQ.md]: Is there a good doc link for this?

File: FAQ.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+If the machine’s status does not have any instance ID or networking information associated with it, most likely the instance has failed to be created.  This can be due to misconfiguration, inadequate or invalid cloud credentials, cloud API quota exhaustion, or other cloud API problems such as an outage or temporary network condition.  You will need to inspect the machine-controllers logs for more definitive information.
+
+## What happens when I delete a Machine?
+This section assumes you’ve deleted a machine from the API, rather than deleting a VM or Instance in the cloud provider.
+
+When a machine is marked deleted, the associated node is immediately cordoned and drained.  Draining utilizes the eviction API, so PodDisruptionBudgets will be respected.  If a node cannot be successfully drained due to a PDB, the operation will retry indefinitely.
+
+After an associated node is drained, the associated instance or VM is deleted from the cloud provider.  After the instance is deleted in the cloud provider, the associated node object is deleted from the API.
```
Comment (JoelSpeed) [inline FAQ.md]: "An" might suggest there can be more than one
```suggestion
After the associated node is drained, the associated instance or VM is deleted from the cloud provider.  After the instance is deleted in the cloud provider, the associated node object is deleted from the API.
```

File: FAQ.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+Finally, the finalizer is removed from the machine object and the machine object is removed from the API.
+
+## I want to skip draining when I delete a Machine
+This is not recommended for most cases, especially Master Machines.  Properly draining machines will respect PodDisruptionBudgets and prevent the cluster and workloads from going into an unhealthy state.
```
Comment (JoelSpeed) [inline FAQ.md]: Do OpenShift docs refer to the machines as `Master` or `Control Plane`, I know upstream they eradicated the use of the term `master` as much as possible, we should try to as well if possible

File: FAQ.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+You can optionally set an **annotation** **"machine.openshift.io/exclude-node-draining"** on each machine object you wish for draining to be skipped.  This can be applied or removed at any time.
```
Comment (JoelSpeed) [inline FAQ.md]: Might be worth clarifying the value of the annotation rather than just the name? (I assume it doesn't matter?)

File: FAQ.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+
+## What happens if I delete an Instance or VM outside of the Machine API, such as in the AWS web console?
+This is not recommended.  By default, the machine-api will not take any corrective action.  If you are utilizing MachineHealthChecking, the machine may get deleted depending on the configuration of the MHC.
```
Comment (JoelSpeed) [inline FAQ.md]: Adding `ing` to the end of resources to make them verbs irks me a bit 😅 Feel free to ignore
```suggestion
This is not recommended.  By default, the machine-api will not take any corrective action.  If you utilize a MachineHealthCheck, the machine may get deleted depending on the configuration of the MHC.
```

Comment (michaelgugino) [reply FAQ.md]: how about 'utilizing MachineHealthChecks' plural?

File: FAQ.md (hunk 5 above)
Comment (michaelgugino) [reply FAQ.md]: Value doesn't matter.  I can clarify.

File: FAQ.md (hunk 6 above)
Comment (JoelSpeed) [reply FAQ.md]: Yeah that's also good :)

File: FAQ.md (hunk 1 above)
Comment (enxebre) [reply FAQ.md]: https://github.com/openshift/enhancements/blob/master/enhancements/machine-api/machine-instance-lifecycle.md

File: FAQ.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+This is not recommended.  This may result in orphaned node objects and orphaned compute resources.
+
+## How do I set a Node’s Role (eg, Worker)
+This is set via the MCO’s config, not the machine-api. The config a machine receives upon boot is decided by the machine's user-data.
+
+## How does the user-data get created for the VMs?
+The user-data is generated by the installer.  The machine-api is just a consumer of the user-data and simply passes the data to the cloud.  Refer to documentation for the machine-config-operator (not part of the machine-api) for creating new machine pools and user-data.
+
+## Adding Labels to Nodes via Machines
+Labels are only added to nodes via machines at creation time.  You can add arbitrary labels which are applied to nodes immediately after they join the cluster.  Adding subsequent labels on the machine object (or in a machineset’s template) will have no effect.
```
Comment (enxebre) [inline FAQ.md]: may be add the api path to the label here, that's usually confusing as there're multiple labels

File: FAQ.md (hunk 4 above)
Comment (michaelgugino) [reply FAQ.md:44]: The 'control plane' consists of 'master machines' according to this: https://docs.openshift.com/container-platform/4.3/architecture/control-plane.html#defining-masters_control-plane

I'm going to stick with current language for now.

**Comment by elmiko** [comment]
i love this, thanks Mike!

/approve

**Pull request merged**